### ISSUE TITLE
go build not use vendor package in 1.13.5

### ISSUE THREAD

**Issue by smileusd** [golang/go#38640]
When i build my project and i change a code in vendor, but i use 
```CGO_ENABLED=0 GOOS=linux go build  -mod=vendor -v -a -o xxx ```
it also not use vendor directory:
```sigs.k8s.io/sig-storage-lib-external-provisioner/controller```
Compare to 1.12.16 the result is :
```github.com/kubernetes-csi/external-provisioner/vendor/sigs.k8s.io/sig-storage-lib-external-provisioner/controller```

macos also the same problem.

Does any other control arguments for this?
I only see the module doc:
```
When using modules, the go command completely ignores vendor directories.

By default, the go command satisfies dependencies by downloading modules
from their sources and using those downloaded copies (after verification,
as described in the previous section). To allow interoperation with older
versions of Go, or to ensure that all files used for a build are stored
together in a single file tree, 'go mod vendor' creates a directory named
vendor in the root directory of the main module and stores there all the
packages from dependency modules that are needed to support builds and
tests of packages in the main module.

To build using the main module's top-level vendor directory to satisfy
dependencies (disabling use of the usual network sources and local
caches), use 'go build -mod=vendor'. Note that only the main module's
top-level vendor directory is used; vendor directories in other locations
are still ignored.
```

### What version of Go are you using (`go version`)?
1.13.5
<pre>
$ go version
`go version go1.13.5 linux/amd64`
`go version go1.13.5 darwin/amd64`
</pre>

### Does this issue reproduce with the latest release?
Sability reproduce

### What operating system and processor architecture are you using (`go env`)?

$ go env
```
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/root/.cache/go-build"
GOENV="/root/.config/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/local/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/linux_amd64"
GCCGO="gccgo"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build141141157=/tmp/go-build -gno-record-gcc-switches"
```

### What did you do?
`go build -mod=vendor -v -a `

### What did you expect to see?
```github.com/kubernetes-csi/external-provisioner/vendor/sigs.k8s.io/sig-storage-lib-external-provisioner/controller```

### What did you see instead?
```sigs.k8s.io/sig-storage-lib-external-provisioner/controller```

**Comment by smileusd** [comment]
Set  `GO111MODULE=off` can fix this problem, but in 1.12.16 i don't set  `GO111MODULE` and i also not find `GO111MODULE` by `go env`. Is it correct situation ? I was confuse for use vendor when in 1.13.5.

**Comment by agnivade** [comment]
Hi @smileusd, unlike other projects, we do not use the issue tracker for questions such as these. It is only used for bugs and feature proposals. I will close this issue, but please feel free to ask it in any of these forums below:

* [Stack Overflow](https://stackoverflow.com/questions/tagged/go) with questions tagged "go"

* [The Go Forum](https://forum.golangbridge.org/), a web-based forum

* [Gophers Slack](https://gophers.slack.com), use the [invite app](https://invite.slack.golangbridge.org/) for access. The `#general` channel is a good starting point.

* [Go Community on Hashnode](https://hashnode.com/n/go) with questions and posts tagged with "go"

* [The golang-nuts mailing list](https://groups.google.com/d/forum/golang-nuts)

* **IRC** channel #go-nuts on Freenode

Thanks
